4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-highlights-script-backtracing.md

---
type: entry
section: highlights
rank: 2
importance: 2
anchor: script-backtracing
title: Script backtracing

> blockquote: How deep the rabbit hole goes?

Reviewer: ```suggestion
blockquote: How deep does the rabbit hole go?
```

Author: I edited the sentence to “You stay in Wonderland, and I show you how deep the rabbit hole goes.”
[That's a quote from Morpheus in the Matrix.](https://web.archive.org/web/20220301025123/https://www.newyorker.com/culture/cultural-comment/the-rabbit-hole-rabbit-hole)

https://youtu.be/zE7PKRjrid4?si=_DkufQD94cbewsws&t=94

text: |
  Using the Release build target has been a blessing and a curse for numerous developers.

>   It permits the game to continue running even if it encountered an unexpected error, enhancing the players' experience. But it prevents the developers to get crucial information that is sorely needed to iron out issues.

Reviewer: ```suggestion
  It permits the game to continue running even if it encountered an unexpected error, enhancing the players' experience. But it prevents the developers getting crucial information that is sorely needed to iron out issues.
```

Author: I removed the sentence/reworked the feature text.

>   It permits the game to continue running even if it encountered an unexpected error, enhancing the players' experience. But it prevents the developers to get crucial information that is sorely needed to iron out issues.

Reviewer 2: That's not a feature of Release build, but Godot in general. If anything, Release is more likely to crash because of some removed debug checks. It's main advantage is the performance.

Author: I updated the text.

>   It permits the game to continue running even if it encountered an unexpected error, enhancing the players' experience. But it prevents the developers to get crucial information that is sorely needed to iron out issues.

Reviewer 3: ```suggestion
  It permits the game to continue running even if it encountered an unexpected error, enhancing the players’ experience. But it prevents the developers getting crucial information that is sorely needed to iron out issues.
```

Author: I removed the paragraph entirely, as I reworked the feature text.

>   That's why we're introducing custom loggers and script backtracing.

Reviewer 3: ```suggestion
  That’s why we’re introducing custom loggers and script backtracing.
```

>   Custom loggers permit developers to intercept log messages and errors. That enables the possibility of creating a bug reporting tool from within your game.

Reviewer 3: ```suggestion
  Custom loggers allow developers to intercept log messages and errors. That enables the possibility of creating a bug reporting tool from within your game.
```
I'd use "allow" here

>   Script backtracing, on the other hand, give developers the exact detail on where an issue happened in the code. Those are available in the Release build as well. (just make sure that the _Debug > Settings > GDScript > Always Track Call Stacks_ project setting is enabled)

Reviewer: ```suggestion
  Script backtracing, on the other hand, give developers the exact detail on where an issue happened in the code. Those are available in the Release build as well. Just make sure that the _Debug > Settings > GDScript > Always Track Call Stacks_ project setting is enabled.
```

Author: I put the sentence in footnote finally.

>   Script backtracing, on the other hand, give developers the exact detail on where an issue happened in the code. Those are available in the Release build as well. (just make sure that the _Debug > Settings > GDScript > Always Track Call Stacks_ project setting is enabled)

Reviewer 3: ```suggestion
  Script backtracing, on the other hand, gives developers the exact details on where an issue happened in the code. Those are available in Release builds as well (just make sure that the _Debug > Settings > GDScript > Always Track Call Stacks_ project setting is enabled).
```

Author: I put (just make sure) in a footnote instead.

---
type: entry
section: highlights
rank: 2
importance: 2
anchor: script-backtracing
title: Script backtracing
blockquote: You stay in Wonderland, and I show you how deep the rabbit hole goes
text: |

>   Those who launched a game using Godot can attest: it is often hard to debug why an error occurs on users devices. Is it a problem caused by the developers code or is it instead a bug from the engine itself?

Reviewer 4: ```suggestion
  Those who launched a game using Godot can attest: it is often hard to debug why an error occurs on users' devices. Is it a problem caused by the developers' code, or a bug from the engine itself?
```

  That’s why we’re introducing custom loggers and script backtracing.

>   Custom loggers allow developers to intercept log messages and errors. That makes it possible to create a bug reporting tool from within your game.

Reviewer: ```suggestion
  Custom loggers allow developers to intercept log messages and errors. That makes it possible to create a tool for reporting bugs within your game.
```

>   Custom loggers allow developers to intercept log messages and errors. That enables the possibility of creating a bug reporting tool from within your game.

Reviewer 3: ```suggestion
  Custom loggers allow developers to intercept log messages and errors. That makes it possible to create a bug reporting tool from within your game.
```
"Enables the possibility" sounds very clunky IMO
